import numpy as np

def fred(a,b):
    c = (a+b)*12
    return c

a = 13.2
b = 3.9

c = fred(a,b)

print("c = ", c)

In [18]:
#rule 1
def rule1(dA, dB):
    dQ = (c*dA)
    return dQ
c = 3
dA = 4



dx = rule1(c,dA)
print (dx)
#rule 2
def rule2(c,m,A,dA):
    dQ = c*m*(A**(m-1))*dA
    return dQ
c = 5
m = 7
A = 10
dA = 12

dx = rule2(c,m,A,dA)
print(dx)
#rule 3
def rule3(dA, dB):
    dQ = np.sqrt(dA**2 + dB**2)
    return dQ

dx1 = 0.01 
dx2 = 0.02

dx = rule3(dx1,dx2)
print (dx)
#rule 4
def rule4(m,A,dA,n,dB,B):
    dQ = np.sqrt(((m*dA)/A)**2+((n*dB)/B)**2)
    return dQ
m = 1
dA = 3
A = 10
n =1 
dB =5
B = 10
dx = rule4(m,dA,A,n,dB,B)
print(dx)

9
420000000
0.022360679774997897
3.370624736026114


In [22]:
#rule 4
def rule4(m,A,dA,n,dB,B):
    dQ = np.sqrt((((m*dA)/A)**2)+(((n*dB)/B)**2))
    return dQ
m = 1
dA = 0.005
A = 0.99
n = 1 
dB = 0.00696
B = 2.1369
dx = rule4(m,A,dA,n,dB,B)
print(dx)

0.006009659392142002


Our uncertainty was 0.006 in last weeks lab and rule 4 calculated 0.006

In [24]:
x = np.array([1.1,1.3,1.4,0.9,0.95,1.05])

avg_x = np.average(x)
err_x = np.std(x)/np.sqrt(3)
print(avg_x,err_x)

1.1166666666666667 0.10363754503432018


The avergae of the data set provided was 1.1667 and the program gave the average of 1.116666666667 an duncertainty of 0.1036


$Q = \sqrt{(\delta A)^2 + (\delta B)^2}$

$Q = \sqrt{(\frac{0.05}{0.99})^2 + (\frac{0.00696}{2.1369})^2}$